src/Intl/Idn/Punycode/Exception/InvalidInputException.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Polyfill\Intl\Idn\Punycode\Exception;
+
+/**
+ * Copyright (c) 2020 Trevor Rowbotham
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * Originally forked from
+ * https://github.com/TRowbotham/punycode/tree/1.0.0
+ *
+ * @author Trevor Rowbotham <trevor.rowbotham@pm.me>
+ */
+class InvalidInputException extends PunycodeException
```

Comment:
why do we have some many custom exception classes here, that are not polyfills and are not internal either ? the error reporting must also match the one of the extension.

Reply:
The `IdnException`, `LookupFailedException`, and `MappingException` will all be going away with the new mapping data. Those were the only exceptions that could propagate to the caller. 

The calls to `Punycode::encode()`/`Punycode::decode()` are all wrapped in `try/catch` statements, so the 3 Punycode excptions would never propagate to the caller. Those exceptions could be replaced by a generic `Exception`, if you would prefer that, since in `Idn` we only ever catch the base exception `PunycodeException` and have no need to differentiate between the different error conditions. The custom exceptions are mostly informative to the reader in this case.